storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh

```diff
@@ -0,0 +1,587 @@
+  echo "$1" | sed -e 's,\([^a-zA-Z0-9/_.=-]\),\\\1,g'
+}
+
+usage () {
+    cat <<EOF
+Usage: $0 [OPTIONS]
+    The following options may be given :
+        --builddir=DIR      Absolute path to the dir where all actions will be performed
+        --get_sources       Source will be downloaded from github
+        --build_src_rpm     If it is 1 src rpm will be built
+        --build_source_deb  If it is 1 source deb package will be built
+        --build_rpm         If it is 1 rpm will be built
+        --build_deb         If it is 1 deb will be built
+        --build_tarball     If it is 1 tarball will be built
+        --install_deps      Install build dependencies(root previlages are required)
+        --branch            Branch for build
+        --repo              Repo for build
+        --help) usage ;;
+Example $0 --builddir=/tmp/PXB --get_sources=1 --build_src_rpm=1 --build_rpm=1
+EOF
+        exit 1
+}
+
+append_arg_to_args () {
+  args="$args "$(shell_quote_string "$1")
+}
+
+ parse_arguments() {
+    pick_args=
+    if test "$1" = PICK-ARGS-FROM-ARGV
+    then
+        pick_args=1
+        shift
+    fi
+
+    for arg do
+        val=$(echo "$arg" | sed -e 's;^--[^=]*=;;')
+        case "$arg" in
+            # these get passed explicitly to mysqld
+            --builddir=*) WORKDIR="$val" ;;
+            --build_src_rpm=*) SRPM="$val" ;;
+            --build_source_deb=*) SDEB="$val" ;;
+            --build_rpm=*) RPM="$val" ;;
+            --build_deb=*) DEB="$val" ;;
+            --get_sources=*) SOURCE="$val" ;;
+            --build_tarball=*) TARBALL="$val" ;;
+            --branch=*) BRANCH="$val" ;;
+            --install_deps=*) INSTALL="$val" ;;
+            --branch=*) BRANCH="$val" ;;
+            --repo=*) REPO="$val" ;;
+            --help) usage ;;
+            *)
+              if test -n "$pick_args"
+              then
+                  append_arg_to_args "$arg"
+              fi
+              ;;
+        esac
+    done
+}
+
+check_workdir(){
+    if [ "x$WORKDIR" = "x$CURDIR" ]
+    then
+        echo >&2 "Current directory cannot be used for building!"
+        exit 1
+    else
+        if ! test -d "$WORKDIR"
+        then
+            echo >&2 "$WORKDIR is not a directory."
+            exit 1
+        fi
+    fi
+    return
+}
+
+add_percona_yum_repo(){
+    if [ ! -f /etc/yum.repos.d/percona-dev.repo ]; then
+        curl -o /etc/yum.repos.d/percona-dev.repo https://jenkins.percona.com/yum-repo/percona-dev.repo
+	    sed -i 's:$basearch:x86_64:g' /etc/yum.repos.d/percona-dev.repo
+    fi
+    return
+}
+
+add_percona_apt_repo(){
+    if [ ! -f /etc/apt/sources.list.d/percona-dev.list ]; then
+        echo "deb http://jenkins.percona.com/apt-repo/ $OS_NAME main" > /etc/apt/sources.list.d/percona-dev.list
+        wget -q -O - http://jenkins.percona.com/apt-repo/8507EFA5.pub | apt-key add -
+  	    wget -q -O - http://jenkins.percona.com/apt-repo/CD2EFD2A.pub | apt-key add -
+    fi
+    return
+}
```

Comment:
please check if we really install any packages from this repo. for example this function was removed in many build scripts

Reply:
Done, removed